src/ast/mod.rs

```diff
@@ -192,6 +215,12 @@ pub enum Expr {
     Identifier(Ident),
     /// Multi-part identifier, e.g. `table_alias.column` or `schema.table.col`
     CompoundIdentifier(Vec<Ident>),
+    /// JsonIdentifier eg: data->'tags'
+    JsonIdentifier {
+        ident: Ident,
```

Comment:
I is there a reason `ident` isn't also a `Box<Expr>`? The examples from the postgres docs seem to show expressions being used as the left hand side (e.g. `'foo'::json`)

Reply:
I missed it. I've made necessary changes and test is added for that.